### DISCUSSION TITLE
broker load 中 无法使用 UDF ？求解答

### DISCUSSION THREAD

**Discussion by limeiyang** [apache/doris#18018] [A - General / Q&A] [answered]
版本：1.2.2

- 问题描述：

可以看到很早 broker load 就支持了 UDF ：https://github.com/apache/doris/pull/4863，但是我在使用的时候却报错：

通过 java 创建的 udf ，`select java_udf_add_one(2);` 这样直接使用的时候可以使用，说明 UDF 创建成功。

1. 但是在 broker load 时（HDFS -> doris）在 SET 中 通过 `a = java_udf_add_one(2)` 报错：
`type:ETL_RUN_FAIL; msg:errCode = 2, detailMessage = No matching function with signature: java_udf_add_one...` 

2. 在 UDF 前面加上数据库名称：`a = databaseName.java_udf_add_one(2)` 报错：No cluster name

**Comment by LemonLiTree** [comment, marked as answer]
Java udf is a new function released last year, and the udf supported by the broker is the native udf, which is no longer recommended
